pkg/placement/controller.go

```diff
@@ -387,12 +411,20 @@ func (c *Controller) processNextWorkItem(ctx context.Context) bool {
 func (c *Controller) reconcile(ctx context.Context, key util.Key) error {
 	var obj runtime.Object
 	var err error
+	// special handling for placement-decision resource as it is the only
+	// resource that is queued directly as key, without necessarily first
+	// existing as an object.
+	if util.KeyIsForPlacementDecision(key) {
```

Comment:
This controller could check for `KeyIsForPlacement` and `KeyIsForCRD` up here too.

Reply:
They both require obj to be set, which is done inside the block below this line. I think this should be refactored in the work after, what do you think?

Reply:
OK with me.